bus_sdk/__init__.py

```diff
@@ -0,0 +1,36 @@
+from bus_sdk.publishers import (
+    EventPublisher,
+    get_publisher_types
+)
+
+
+__version__ = '0.0.1'
+
+
+DEFAULT_NAME = 'default_publisher'
+DEFAULT_TYPE = 'sns'
+PUBLISHERS = {}
+PUBLISHER_TYPES = get_publisher_types()
+
+
+def get_publisher(pub_name: str = DEFAULT_NAME, pub_type: str = DEFAULT_TYPE,
+                  config: dict = {}) -> EventPublisher:
+    """
+    Publisher factory function.
+
+    If a publisher already exists with the `pub_name` value
+    then you will be returned that instance, otherwise the publisher
+    will be created and returned.
```

Comment:
Service location? Would rather see DI...

Reply:
Not sure what you mean here. 🤔

Reply:
Have left this as is to handle not having to configure the same `publisher` in multiple modules (some other projects are more function based and this would avoid some duplication). Though you can still use this to return a new instance which can then be directly injected.